### PR TITLE
HPCC-22787 Return log field names when returning log file content

### PR DESCRIPTION
**Pull request by wangkx**
Also add new option IncludeLogFieldNameLine. If the option is set
to be false, the new Log Field Name line will be removed from the
log file content. The feature is needed by ECLWatch UI.

Fix a bug in readLastNRowsToArray().

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by wangkx** [comment]
@afishbeck please review.

File: esp/scm/ws_topology.ecm [hunk 1]
```diff
@@ -532,6 +532,7 @@ ESPrequest [nil_remove] TpLogFileRequest
     [min_ver("1.05")] bool Zip(false);
     [min_ver("1.05")] int PageNumber(0);
     [min_ver("1.05")] bool LoadData(false);
+    [min_ver("1.29")] bool IncludeLogFieldNameLine(true);
```
Comment (afishbeck) [inline esp/scm/ws_topology.ecm]: Does the caller need to know it is grabbed from a line?  That is would just "IncludeLogFieldNames" be a better name?

File: esp/services/ws_topology/ws_topologyService.cpp [hunk 2]
```diff
@@ -839,13 +861,28 @@ void CWsTopologyEx::readLastNRowsToArray(const char* logName, OwnedIFileIO rIO,
             finger--;
         }
 
-        if (nRowsLogged || (readFrom == 0))
+        if (nRowsLogged)
             break;
 
+        if (readFrom == readLogReq.readLogFrom)
+        {
+            //This is the first chuck of the log file. We need to include the first
```
Comment (afishbeck) [inline esp/services/ws_topology/ws_topologyService.cpp]: "chunk"

File: esp/services/ws_topology/ws_topologyService.hpp [hunk 3]
```diff
@@ -57,6 +57,9 @@ struct ReadLog
     unsigned logfields;
     unsigned columnNumDate;
     unsigned columnNumTime;
+    StringArray logFieldNames;
+    bool includeLogFieldNameLine;
```
Comment (afishbeck) [inline esp/services/ws_topology/ws_topologyService.hpp]: Good habit to start initializing new variables here within class or struct declaration.  Just new ones as part of a PR like this.

**Review by afishbeck — COMMENTED** [review]
@wangkx very minor comments.  Otherwise looks fine.

**Review by afishbeck — APPROVED** [review]
@wangkx looks good.

**Comment by richardkchapman** [comment]
@wangkx If this needs to go into 7.6.0, please retarget to candidate-7.6.0

**Comment by wangkx** [comment]
@richardkchapman the base branch is checnged to candidate-7.6.0.

**Pull request merged**